ecl/eclcc/WORKUNITS.rst

```diff
@@ -0,0 +1,862 @@
+of how the generated code is structured and what part it plays in executing a query.
+
+Note the term "Query" is used as a generic term to cover read-only queries (typically run in roxie) and ETL style
+queries which create lots of persistent datafiles (typically run in Thor).  Also, the term "workunit" is used a bit
+ambiguously.  The dll created from a query is called a workunit (which is static), but "workunit" is also used to
+describe a single execution of a query (which includes the parameters and results).  It should be clear from the
+context which of these is meant.
+
+Contents of a workunit
+======================
+
+A workunit is generated by the ecl compiler, and consists of a single dll/shared object.  That dll contains
+several different elements:
+
+* Various c++ helper classes, and exported factory methods are used to create instances of those classes.
+* An XML resource containing different pieces of information about a workunit, including workflow and graphs.
+* Other user-defined resources included in the manifest.
+
+The workunit dll contains everything that the engines need to execute the query.  When a workunit is executed, key
+elements of the xml information are cloned from the workunit dll and copied into a database (currently dali).  This
+is then extended with other information - e.g., input parameters, results, various statistics.  The IWorkUnit interface
+(define in common/workunit/workunit.hpp) is the public interface onto the contents, and hides many of the
+implementation details.
+
+How is it used?
+===============
+
+The workunit information is used by most of the components in the system.  The following is a quick outline:
+
+* eclcc
+  Creates a workunit dll from an ecl query.
+* eclccserver
+  Executes eclcc to create a workunit dll, and then clones some of the information into dali to create an active instance, ready to execute.
+* esp
+  Uses information in the workunit to publish workunits - including details of what parameters it takes, how they should be formatted, and what results it returns.
+* eclscheduler
+  Monitors workunits that are waiting for events, and updates them when those events occur.
+* eclagent/roxie
+  Process the different workflow actions
+* hthor/roxie/Thor
+  Execute graphs
+  
+Example
+*******
+
+The following ECL will be used as an example for the rest of the discussion.  It is a very simple search that 
+takes a string parameter 'searchName', which is the name of the person to search for, and returns the matching records
+from an index.  It also outputs the word 'Done!' as a separate result.
+
+::
+
+    STRING searchName := 'Smith' : STORED('searchName');
+    nameIndex := INDEX({ STRING40 name, STRING80 address }, 'names');
+    results := nameIndex(KEYED(name = searchName));
+    OUTPUT(results);
+    OUTPUT('Done!');
+
+Extracts from the XML and C++ that are generated from this example will be included in the following discussion.
```

Comment:
You should explain here that 'eclcc -wu' generates the XML workunit so  that they may see for themselves.
I would also add here -fsaveCppTempFiles generates the .cpp files, I know you mention this later but it should be mentioned here first.